### ISSUE TITLE
Read Wichmann et al

### ISSUE THREAD

**Issue by rockladyeagles** [vgcagle/DARC#5]
When you're done, write your takeaways from the paper in a comment on this issue.

**Comment by rockladyeagles** [comment]
The main benefit specific to the supply chain project is that companies who have visibility into where they "fit" in the overall supply chains, so that they can anticipate delays and disruptions before they occur.

**Comment by rockladyeagles** [comment]
State of the art: most companies have no more information than their immediate graph neighbors.

> "Any company knows its direct suppliers and customers, yet already knowledge of second-tier suppliers tends to be incomplete."

**Comment by rockladyeagles** [comment]
> "Conceptually, the extraction process requires two stages: First, NER: mentions of named entities need to be detected and classified as organisations (as opposed to locations, persons etc.). Secondly, Relation Extraction: each pair of two organisational mentions needs to be classified with respect to the stated relationship between them.

This paper's project uses an existing solution for the first stage (spaCy, CoreNLP, and Flair, to form an ensemble) and thus only really implements and tests the second stage.

**Comment by rockladyeagles** [comment]
Organization names themselves are masked before being given to the classifier, so that the classifier was forced to learn general rules rather than memorize specific supplier relationships between specific companies.

**Comment by rockladyeagles** [comment]
Uses a BiLSTM network with pre-trained GloVe embeddings (in 300 dimensions).
Also, for comparison, does the same task with a simple MLP and a simple SVM.

**Comment by vgcagle** [comment]
The horizontal structure refers to the number of tiers across the supply chain. The vertical structure refers to the number of suppliers or customers represented within each tier (Lambert and Cooper 2000). The term “upstream” is used to denote the direction towards to original supplier whereas “downstream” refers to the direction towards the ultimate customer.

Structural supply chain visibility is often limited. Suppliers have an incentive not to disclose their own supply network to their customers.

Highlights the importance of directionality “company A supplies company B”, and (b) “company A is supplied by company B”. 

Company names in each sentence were automatically masked so that the classifier did not learn relations between specific organizational named entities but between any text sequences tagged as organizations.

The BiLSTM has an embedding layer and considers 16 features in the hidden state of the LSTM layer. It also uses a dropout with a probability of 0.5. The last layer is a dense layer with 5 output units, one for each class. The LSTM model used the standard hyperbolic tangent function (“tanh”) activation function. 
Interesting to look at the comparison of the BiLSTM to the MLP and SVM.

**Comment by harmonyrose** [comment]
The aim of this research was to test the feasibility of automatically generating supply chain maps from text. This is an important field of study because many companies have no supply chain visibility beyond their direct suppliers. Information about extended suppliers helps manage efficiency, resilience, and sustainability. It can also help prevent modern slavery, and can mitigate cascading risks. 

One major difference I noticed between their research and what we will be doing is that their study used sentence-level NER, ignoring coreference resolution. Our work will be document-level, thus creating the need to use coreference resolution.

They used human annotators to create a gold-standard corpus on relation classification. They tested an SVM model (using bag-of-words), an MLP model (using GloVe embeddings), and a BiLSTM model (also using GloVe embeddings). The F1 scores of all 3 models were within 0.04 of each other, which I found surprising. The average F1 score was about ~0.7, demonstrating that the models performed better than dummy classifiers, but there is still room for improvement.

They found that there is a trade-off between the number of relationship classes and the simplicity of the classification task. Having more classes leads to a lower quality program. They used 5 different classes (by "classes" they mean relations, like "A supplies B").

**Comment by rpersing** [comment]
For some sentences, the set of words is identical and only the order of the entity mentions in the sentence is different.

Models that are able to consider word sequences will outperform models that are not able to do so.

They mention GPT-2 and BERT as two models that could provide interesting and compelling results.

"We will ignore relations that would require the resolution of pronouns to company names (co-reference resolution). Furthermore, we will ignore more multi-faceted relations at this stage, such as extracting supplied products or further companies involved". They don't use coreference resolution. They also aren't worried about things like what products are being supplied or if multiple companies are involved in one supply chain. They characterize it as a single-label multi-class classification problem.
